realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -166,6 +167,7 @@ public void removeChangeListener(RealmChangeListener<? extends BaseRealm> listen
      * @throws UnsupportedOperationException if the required RxJava framework is not on the classpath.
      * @see <a href="https://realm.io/docs/java/latest/#rxjava">RxJava and Realm</a>
      */
+    @OptionalAPI(dependencies = {"rx.Observable"})
```

Comment:
For type safety I would change the annotation to use `Class` instead, since the RxJava dependency is `provided` in `realm-library`
```groovy
provided 'io.reactivex:rxjava:1.1.0'
```


```java
@Retention(RetentionPolicy.CLASS)
@Target(ElementType.METHOD)
public @interface OptionalAPI {
    Class[] dependencies();
}
```

now you can use it like this 

```java
@OptionalAPI(dependencies = rx.Observable.class)
    public Observable<RealmResults<E>> asObservable() {
   //...
```